In [1]:
# Load sql extension
%load_ext sql

In [2]:
# Connecting to my sql_project database
%sql mysql://USERNAME:PASSWORD@HOST/DATABASE

'Connected: admin@sql_project'

# Part 1: 5 Exploratory SQL Queries

1. The purpose of the code below is to better understand the price range of the best selling products on skims.com. Specifically, it looks further into the average price of all current best selling items.

In [14]:
%%sql

SELECT ROUND(AVG(price), 2) AS avg_price_of_best_sellers
FROM inventory;

 * mysql://admin:***@isba-dev-01.cwmgtmuif5sr.us-east-1.rds.amazonaws.com/sql_project
1 rows affected.


[(Decimal('42.42'),)]

I discovered that the average price of all current top products is roughly 42.42 dollars
It is important to compare the current average to the average of previous best selling products. The company is still relatively new (less than 4 years since it launched), so it would be interesting to see how this average has changed since the beginning. This average tells us that customers are willing to spend roughly 40 dollars on a garment when shopping at skims.

2. To further look into the price range of the best sellers, this next code will find the highest price of all products on the "best selling" page on skims.com as well as the specific product.

In [21]:
%%sql

SELECT 
    name, 
    MAX(price) AS max_price_best_sellers
FROM inventory
GROUP BY name
ORDER BY max_price_best_sellers DESC
LIMIT 1;

 * mysql://admin:***@isba-dev-01.cwmgtmuif5sr.us-east-1.rds.amazonaws.com/sql_project
1 rows affected.


[('SWIM ZIP FRONT SLEEVELESS ONE PIECE', Decimal('94.00'))]

I discovered that the 'SWIM ZIP FRONT SLEEVELESS ONE PIECE' is the most expensive product out of all the best sellers. 
It costs 94 dollars. This is quite costly for a single swimsuit and shows that customers are willing to pay a premium for a Skims swim suit. Since this swim suit is considered a best seller, and customers are willing to pay over 50 dollars than the average for it, Skims should consider to continue to release more swim suits.

3. The purpose of the code below is to look into the cheapest product from the best sellers as well as the exact price. 

In [30]:
%%sql

SELECT 
    name, 
    MIN(price) AS min_price_best_sellers
FROM inventory 
GROUP BY name 
ORDER BY min_price_best_sellers ASC 
LIMIT 1;

 * mysql://admin:***@isba-dev-01.cwmgtmuif5sr.us-east-1.rds.amazonaws.com/sql_project
1 rows affected.


[('FITS EVERYBODY MICRO THONG', Decimal('14.00'))]

This code shows that the 'FITS EVERYBODY MICRO THONG' is the cheapest product sold out of all best sellers. It costs 14 dollars and can be seen as quite pricey for a single underwear. Customers can easily buy a multi-pack of underwear for the same price at other stores/brands, yet, Skims customers are willing to pay more to purchase Skims underwear. Skims should continue to develop additional types of underwear to continue to drive sales.

4. The purpose of the code below is to compare the total reviews of the two products shown above. This will show which product has more reviews and if the cost factors into how many reviews the product gets.

In [33]:
%%sql

SELECT 
    i.product_id,
    name,
    reviews AS 'total_reviews'
FROM inventory i 
JOIN reviews r
    ON i.product_id = r.product_id
WHERE i.product_id = '9'
    OR i.product_id = '35';

 * mysql://admin:***@isba-dev-01.cwmgtmuif5sr.us-east-1.rds.amazonaws.com/sql_project
2 rows affected.


[(9, 'SWIM ZIP FRONT SLEEVELESS ONE PIECE', 106),
 (35, 'FITS EVERYBODY MICRO THONG', 1504)]

The results above compares the total reviews between the most expensive and cheapest product from the best sellers. We can see that the most expensive product (Swim One Piece) only has 106 reviews while the cheapest product (Micro Thong) has over 1,500 reviews. This also shows that more people are purchasing the underwear compared to the swim suit. Skims should continue to produce underwear as it generates a lot of sales.

5. The purpose of the code below is to analyze how customers rate the two products. Although the Swim One Piece has significantly less reviews than the Micro Thong, it will be interesting to see what the average rating is for both.

In [38]:
%%sql

SELECT 
    i.product_id,
    name,
    rating AS 'avg_rating'
FROM inventory i 
JOIN ratings r
    ON i.product_id = r.product_id
WHERE i.product_id = '9'
    OR i.product_id = '35';

 * mysql://admin:***@isba-dev-01.cwmgtmuif5sr.us-east-1.rds.amazonaws.com/sql_project
2 rows affected.


[(9, 'SWIM ZIP FRONT SLEEVELESS ONE PIECE', 4.5),
 (35, 'FITS EVERYBODY MICRO THONG', 4.6)]

The results above show that the two products have relatively similar average ratings. This shows that customers are satisfied with the overall quality of these two products. It would be best to further look into the customer feedback and reviews to identify any issues or ways to improve the product. This would also help Skims to see what the customers do like and continue to implement the factors the customers enjoy.

## Part 2

1. Which best selling products have the highest customer average ratings? Although they are all best selling items on skims.com, it is important to know which are the most popular within this list.

Business Justification: Understanding which best selling products have the highest ratings will allow the company to make strategic decisions related to product development, marketing and inventory management. By gaining insight on which products are most highly rated by customers, the company can focus on promoting and improving these products while phasing out or improving underperforming products.

SQL Features: JOIN, CASE, CTE

In [5]:
%%sql

WITH products AS (
    SELECT
        c.product_id,
        c.name AS product_name,
        c.category,
        r.rating
    FROM category c
    JOIN inventory i 
        ON c.product_id = i.product_id
    LEFT JOIN ratings r 
        ON c.product_id = r.product_id
)
SELECT 
    product_name, 
    category,
    rating,
    CASE
        WHEN rating = 0 THEN 'No ratings yet'
        WHEN rating >= 4.6 THEN 'Highly rated'
        WHEN rating >= 4 THEN 'Moderately rated'
        ELSE 'Poorly rated'
    END AS rating_level
FROM products
ORDER BY rating DESC;


 * mysql://admin:***@isba-dev-01.cwmgtmuif5sr.us-east-1.rds.amazonaws.com/sql_project
52 rows affected.


[('SOFT LOUNGE TANK', 'SOFT LOUNGE', 4.8, 'Highly rated'),
 ('SOFT LOUNGE SLIP DRESS', 'SOFT LOUNGE', 4.8, 'Highly rated'),
 ('SCULPTING BODYSUIT W. SNAPS', 'SCULPTING', 4.8, 'Highly rated'),
 ('SOFT SMOOTHING BRALETTE', 'SOFT SMOOTHING', 4.8, 'Highly rated'),
 ('SOFT SMOOTHING T-SHIRT', 'SOFT SMOOTHING', 4.8, 'Highly rated'),
 ('FITS EVERYBODY THONG', 'FITS EVERYBODY', 4.8, 'Highly rated'),
 ('SOFT LOUNGE LONG SLIP DRESS', 'SOFT LOUNGE', 4.8, 'Highly rated'),
 ('COTTON JERSEY DIPPED THONG', 'COTTON', 4.8, 'Highly rated'),
 ('COTTON JERSEY CHEEKY TANGA', 'COTTON', 4.8, 'Highly rated'),
 ('FITS EVERYBODY DIPPED FRONT THONG', 'FITS EVERYBODY', 4.8, 'Highly rated'),
 ('FITS EVERYBODY SCOOP BRALETTE', 'FITS EVERYBODY', 4.7, 'Highly rated'),
 ('FITS EVERYBODY T-SHIRT BODYSUIT', 'FITS EVERYBODY', 4.7, 'Highly rated'),
 ('FITS EVERYBODY HIGH NECK BODYSUIT', 'FITS EVERYBODY', 4.7, 'Highly rated'),
 ('FITS EVERYBODY T-SHIRT BRA', 'FITS EVERYBODY', 4.7, 'Highly rated'),
 ('FITS EVERYBODY CHEEKY BRIEF', 'FITS EVERYBODY', 4.7, 'Highly rated'),
 ('FITS EVERYBODY CROSSOVER BRALETTE', 'FITS EVERYBODY', 4.7, 'Highly rated'),
 ('FITS EVERYBODY DIPPED FRONT THONG', 'FITS EVERYBODY', 4.7, 'Highly rated'),
 ('FITS EVERYBODY LONG SLEEVE CREW NECK BODYSUIT', 'FITS EVERYBODY', 4.7, 'Highly rated'),
 ('COTTON JERSEY BOY SHORT', 'COTTON', 4.7, 'Highly rated'),
 ('FITS EVERYBODY BOY SHORT', 'FITS EVERYBODY', 4.7, 'Highly rated'),
 ('COTTON JERSEY LONG SLEEVE T-SHIRT', 'COTTON', 4.7, 'Highly rated'),
 ('COTTON JERSEY T-SHIRT', 'COTTON', 4.7, 'Highly rated'),
 ('COTTON RIB LEGGING', 'COTTON', 4.7, 'Highly rated'),
 ('SOFT LOUNGE BOXER', 'SOFT LOUNGE', 4.7, 'Highly rated'),
 ('SOFT LOUNGE LONG SLEEVE DRESS', 'SOFT LOUNGE', 4.7, 'Highly rated'),
 ('FITS EVERYBODY CROSSOVER BRALETTE', 'FITS EVERYBODY', 4.6, 'Highly rated'),
 ('FITS EVERYBODY MICRO THONG', 'FITS EVERYBODY', 4.6, 'Highly rated'),
 ('FITS EVERYBODY T-SHIRT BODYSUIT', 'FITS EVERYBODY', 4.6, 'Highly rated'),
 ('FITS EVERYBODY SCOOP BRALETTE', 'FITS EVERYBODY', 4.6, 'Highly rated'),
 ('COTTON RIB THONG', 'COTTON', 4.6, 'Highly rated'),
 ('SWIM DIPPED TIE BOTTOMS', 'SWIM', 4.6, 'Highly rated'),
 ('COTTON JERSEY FULL BRIEF', 'COTTON', 4.5, 'Moderately rated'),
 ('COTTON JERSEY TRIANGLE BRALETTE', 'COTTON', 4.5, 'Moderately rated'),
 ('SWIM SCOOP NECK ONE PIECE', 'SWIM', 4.5, 'Moderately rated'),
 ('COTTON RIB BOXER', 'COTTON', 4.5, 'Moderately rated'),
 ('SWIM TANK BIKINI TOP', 'SWIM', 4.5, 'Moderately rated'),
 ('SCULPTING BRALETTE', 'SCULPTING', 4.5, 'Moderately rated'),
 ('SWIM ZIP FRONT SLEEVELESS ONE PIECE', 'SWIM', 4.5, 'Moderately rated'),
 ('SOFT SMOOTHING SHORT', 'SOFT SMOOTHING', 4.5, 'Moderately rated'),
 ('COTTON JERSEY T-SHIRT BRA', 'COTTON', 4.4, 'Moderately rated'),
 ('SWIM TRIANGLE TOP', 'SWIM', 4.4, 'Moderately rated'),
 ('BOYFRIEND LONG SLEEVE T-SHIRT', 'BOYFRIEND', 4.3, 'Moderately rated'),
 ('COTTON RIB TANK', 'COTTON', 4.3, 'Moderately rated'),
 ('FITS EVERYBODY SQUARE NECK BODYSUIT', 'FITS EVERYBODY', 4.3, 'Moderately rated'),
 ('SCULPTING THONG BODYSUIT', 'SCULPTING', 4.2, 'Moderately rated'),
 ('FITS EVERYBODY TRIANGLE BRALETTE', 'FITS EVERYBODY', 4.2, 'Moderately rated'),
 ('SCULPTING BODYSUIT MID THIGH W. OPEN GUSSET', 'SCULPTING', 4.2, 'Moderately rated'),
 ('SOFT SMOOTHING THONG', 'SOFT SMOOTHING', 4.2, 'Moderately rated'),
 ('COTTON PLUNGE BRALETTE', 'COTTON', 4.2, 'Moderately rated'),
 ('FITS EVERYBODY BANDEAU', 'FITS EVERYBODY', 4.1, 'Moderately rated'),
 ('BOYFRIEND T-SHIRT', 'BOYFRIEND', 3.9, 'Poorly rated'),
 ('BOYFRIEND BOXER', 'BOYFRIEND', 3.9, 'Poorly rated')]

In [23]:
%%sql

# Shows the categories total count of "Highly rated" rating_levels
WITH products AS (
    SELECT
        c.product_id,
        c.name AS product_name,
        c.category,
        r.rating,
        rv.reviews,
        CASE
            WHEN r.rating = 0 THEN 'No ratings yet'
            WHEN r.rating >= 4.6 THEN 'Highly rated'
            WHEN r.rating >= 4 THEN 'Moderately rated'
            ELSE 'Poorly rated'
        END AS rating_level
    FROM category c
    JOIN inventory i 
        ON c.product_id = i.product_id
    LEFT JOIN ratings r 
        ON c.product_id = r.product_id
    LEFT JOIN reviews rv 
        ON r.product_id = rv.product_id
)
SELECT 
    category,
    SUM(CASE WHEN rating_level = 'Highly rated' THEN 1 ELSE 0 END) AS count_highly_rated
FROM products
GROUP BY category
ORDER BY count_highly_rated DESC;


 * mysql://admin:***@isba-dev-01.cwmgtmuif5sr.us-east-1.rds.amazonaws.com/sql_project
7 rows affected.


[('FITS EVERYBODY', Decimal('15')),
 ('COTTON', Decimal('7')),
 ('SOFT LOUNGE', Decimal('5')),
 ('SOFT SMOOTHING', Decimal('2')),
 ('SWIM', Decimal('1')),
 ('SCULPTING', Decimal('1')),
 ('BOYFRIEND', Decimal('0'))]

Insight: The results above shows every product on the best sellers page. It shows the specific category that the product is in, the average rating, as well as the rating's level. A 4.6 rating and above is considered to be highly rated. A rating between 4.0 and 4.5 is considered to be moderately rated. Anything below a 4.0 rating is considered to be poorly rated. Out of the 52 best sellers, 31 are highly rated, 19 are moderately rated, and 2 are poorly rated. The second set of results show the total count of high ratings per category. Fits Everybody has the most 4.6+ ratings out of all categories with a total of 15.

Recommendation: A recommendation is to potentially discontinue the Boyfriend T-Shirt and Boyfriend Boxer as these are the only two products that are poorly rated. It may not be worth trying to improve a product that has low ratings. Instead, they should focus on trying to improve all of the products that are moderately rated and try to bring these ratings up to be within the 4.6 - 5.0 ratings.

2. Which categories/lines are the most popular based on the total reviews? We can infer that more reviews mean more orders were made per category. We will rank the categories from most popular to least popular.

Business Justification: Ranking the different product "lines" or "categories" offered at Skims will allow the company to understand which categories perform the best amongst customers based off reviews. This will allow the company to better plan their next collections and understand which products to continue offering and which to discontinue.

SQL Features: JOIN, GROUP BY, WINDOW FUNCTION

In [43]:
%%sql

SELECT 
    c.category, 
    SUM(r.reviews) AS total_reviews,
    RANK() OVER (ORDER BY SUM(r.reviews) DESC) AS category_rank
FROM 
    category c 
JOIN reviews r 
    ON c.product_id = r.product_id 
GROUP BY 
    c.category 
ORDER BY 
    total_reviews DESC;


 * mysql://admin:***@isba-dev-01.cwmgtmuif5sr.us-east-1.rds.amazonaws.com/sql_project
7 rows affected.


[('FITS EVERYBODY', Decimal('27880'), 1),
 ('COTTON', Decimal('7438'), 2),
 ('SOFT LOUNGE', Decimal('6271'), 3),
 ('BOYFRIEND', Decimal('4138'), 4),
 ('SOFT SMOOTHING', Decimal('3026'), 5),
 ('SCULPTING', Decimal('2955'), 6),
 ('SWIM', Decimal('841'), 7)]

Insight: The results above show the top seven popular categories at Skims. We can interpret that these are the most popular categories as they are the categories listed on their best sellers page. The Fits Everybody line is the most popular line, followed by Cotton, Soft Lounge, Boyfriend, Soft Smoothing, Scultpting, and Swim. Fits Everybody is their oldest line which makes sense that it has the most reviews. More customers have purchased this line and are more familiar with it than the Swim line. The Swim line is the newest addition to Skims and has the least reviews most likely because customers have not had a chance to purchase it or review it yet. 

Recommendation: A recommendation is to continue releasing more items within the Fits Everybody line as it has over 20,000 more reviews than the second most popular category, Cotton. It most likely generates the most revenue for the company, so continuing this line makes most sense. Additionally, they should invest in marketing the Swim line as customers may not know that they are now carrying swim wear. If the marketing is successful, it can also generate more sales and bring exposure to their new line.

3. How does cost affect the categories popularity? 

Business Justification: Looking into the average price of all products per category will help indicate why certain categories are more popular than others. We already saw the category popularity based off total reviews (orders), but viewing this based off the price factor is also crucial to truly get a better understanding. 

SQL Features: JOIN, GROUP BY, VIEW, SUBQUERY

In [15]:
%%sql 

# Creating the view
CREATE VIEW category_avg_price AS
SELECT 
    c.category, 
    ROUND(AVG(i.price), 2) AS avg_price
FROM category c 
JOIN inventory i 
    ON c.product_id = i.product_id 
JOIN reviews r 
    ON c.product_id = r.product_id 
WHERE 
    i.price <= 
    (
        SELECT AVG(price) 
        FROM inventory 
        WHERE product_id IN 
    (
        SELECT product_id 
        FROM category 
        WHERE category=c.category)
    )
GROUP BY c.category
ORDER BY avg_price;

 * mysql://admin:***@isba-dev-01.cwmgtmuif5sr.us-east-1.rds.amazonaws.com/sql_project
0 rows affected.


[]

In [21]:
%%sql

# Query using the View to compare the categories average prices with the total reviews
SELECT 
    c.category, 
    c.total_reviews, 
    v.avg_price
FROM 
    (SELECT 
        category, 
        SUM(reviews) AS total_reviews
    FROM 
        category c 
    JOIN reviews r 
        ON c.product_id = r.product_id 
    GROUP BY 
        category) c
JOIN 
    category_avg_price v 
    ON c.category = v.category 
ORDER BY 
    total_reviews DESC;

 * mysql://admin:***@isba-dev-01.cwmgtmuif5sr.us-east-1.rds.amazonaws.com/sql_project
7 rows affected.


[('FITS EVERYBODY', Decimal('27880'), Decimal('25.67')),
 ('COTTON', Decimal('7438'), Decimal('23.14')),
 ('SOFT LOUNGE', Decimal('6271'), Decimal('36.00')),
 ('BOYFRIEND', Decimal('4138'), Decimal('28.00')),
 ('SOFT SMOOTHING', Decimal('3026'), Decimal('24.00')),
 ('SCULPTING', Decimal('2955'), Decimal('38.00')),
 ('SWIM', Decimal('841'), Decimal('40.67'))]

Insight:  The results show the total average price of all products per category. Out of the seven categories, Cotton has the lowest total average price, followed by Soft Smoothing, Fits Everybody, Boyfriend, Soft Lounge, Sculpting, and Swim. Fits Everybody has the most reviews (orders) and high customer ratings out of all categories which explained its popularity. We wanted to see if the price was the main factor as to why it was popular, however, it is the third cheapest overall category. This proves that customers genuinely like this category for other reasons simply than just the price. Swim was the least popular category when it came to total reviews and also is the most expensive category which may be due to how much higher the average product costs are compared to other cateogories.

Recommendation: While it may be risky, raising the price of the Fits Everybody products may help the company to generate more revenue from this category. We saw that this line is a customer favorite and not because of the price. Customers most likely will continue to purchase their favorites from this collection even if the price increases. Also, Skims should consider lowering the average price of the Swim products to see if it will generate more sales. Customers most likely are not purchasing the swim products due to the high price. 